android/KMEA/app/src/main/java/com/keyman/engine/KMManager.java

```diff
@@ -2416,11 +2416,8 @@ public static void applyKeyboardHeight(Context context, int height, int orientat
       height = defaultHeightForContext;
     } else {
       // Applying gating to 50%-200% of default height (following Keyman)
-      if (height < (defaultHeightForContext / 2)) {
-        height = (int) (defaultHeightForContext / 2);
-      } else if (height > (defaultHeightForContext * 2)) {
-        height = (int) (defaultHeightForContext * 2);
-      }
+      height = Math.max((int) getKeyboardHeightMin(context), height);
+      height = Math.min((int) getKeyboardHeightMax(context), height);
```

Comment:
When getKeyboardHeightMin() or getKeyboardHeightMax() return `KMManager.KeyboardHeight_Reset` (= 0), should we reset height to the default (and not apply gating to 0)?

Reply:
I think only positive heights get passed to the KeymanWeb engine
https://github.com/keymanapp/keyman/blob/9c9696c5fdcf2e90c7c87ffa3bfc744263bf82c2/android/KMEA/app/src/main/assets/android-host.js#L160-L168

Reply:
I don't think KMManager.KeyboardHeight_Reset should ever be returned from those functions. It makes no sense -- those are supposed to be essentially constant min/max values and returning KMManager.KeyboardHeight_Reset just means complications in usage. This reflects the more general discussion (can of worms) I opened up at https://github.com/keymanapp/keyman/pull/13663/files#r2559006329